connman/plugins/ethernet.c

```diff
@@ -73,7 +73,7 @@ static int get_vlan_vid(const char *ifname)
 		return -errno;
 
 	vifr.cmd = GET_VLAN_VID_CMD;
-	stpncpy(vifr.device1, ifname, sizeof(vifr.device1));
+	stpncpy(vifr.device1, ifname, sizeof(vifr.device1) - 1);
```

Comment:
Ed. note: I think this is a wrong fix to a non-issue (kernel does null terminate these), but it's from upstream, so let's just roll with it. IMO correct fix would be to err out if the name does not fit in the buffer.

Reply:
Yeah... well, we can still drop this and propose a proper fix to upstream (?) as it does have the same issue regarding hot-plugs afaik.

Reply:
Or, well, lets keep it the same as upstream. We'll fix it later if this causes  any issues, not that it really should but.